### PR TITLE
setup script for running binskim

### PR DESCRIPTION
**Pull request by MS-TimothyMothra**
BinSkim will fail on our test directories.
This script will copy necessary files to a new directory to be scanned

https://www.microsoft.com/en-us/SDL/adopt/tools.aspx

### CODE REVIEW

File: SDL/BinSkimSetup.ps1 [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿[CmdletBinding()]
+Param(
+    [Parameter(Mandatory=$True)]
+    [string]$buildDirectory,
+    
+    [Parameter(Mandatory=$True)]
+    [string]$binSkimDirectory
+)
+
+#$buildDirectory = "C:\Repos\SDK\bin\Release\Src\"
```
Comment (d3r3kk) [inline SDL/BinSkimSetup.ps1]: Suggestion: 
You can add default values to the parameters to make testing easier. Perhaps something like:

    [string]$buildDirectory=(Join-Path -Path (Split-Path -parent $PSCommandPath) -ChildPath "bin\Release\Src")

...might make testing locally a bit more handy.

File: SDL/BinSkimSetup.ps1 [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#$binSkimDirectory = "C:\Repos\SDK\binSkim\"
+
+Write-Host "`nPARAMETERS:";
```
Comment (d3r3kk) [inline SDL/BinSkimSetup.ps1]: Consider using 

    Write-Verbose "Parameters:"

here. Newline not really needed.

File: SDL/BinSkimSetup.ps1 [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Write-Host "`tbuildDirectory:" $buildDirectory;
+Write-Host "`tbinSkimDirectory:" $binSkimDirectory;
+
+# don't need to clean folder on build server, but is needed for local dev
+if (Test-Path $binSkimDirectory) { Remove-Item $binSkimDirectory -Recurse; }
```
Comment (d3r3kk) [inline SDL/BinSkimSetup.ps1]: Adding the following to your Test-Path will help out a bit in some cases:

    if (Test-Path -Path $binSkimDirectory -ErrorAction SilentlyContinue) { . . . }

File: SDL/BinSkimSetup.ps1 [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# copy all
+Copy-Item -Path $buildDirectory -Filter "*.dll" -Destination $binSkimDirectory -Recurse;
+
+# delete test directories
+Get-ChildItem -Path $binSkimDirectory -Recurse -Directory | 
+ Where-Object {$_ -match "Test"} |
+ Remove-Item -Recurse;
```
Comment (d3r3kk) [inline SDL/BinSkimSetup.ps1]: Do you not need a '-Force' switch on this Remove-Item?

File: SDL/BinSkimSetup.ps1 [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# summary for log output (file list and count)
+Write-Host "`nCOPIED FILES:";
+
+$count = 0;
```
Comment (d3r3kk) [inline SDL/BinSkimSetup.ps1]: Here's a shortcut:

    (Get-ChildItem -Path $binSkimDirectory -Recurse -File).Count

File: SDL/BinSkimSetup.ps1 (hunk 2 above)
Comment (MS-TimothyMothra) [reply SDL/BinSkimSetup.ps1]: I understand the purpose of the Verbose distinction, but in this use case I don't think we want this. If a build fails, we would have to run it a second time with the verbose flag to get the details. 
I would rather have the logging always enabled.
Please let me know if you're considering something I'm not thinking of :)

File: SDL/BinSkimSetup.ps1 (hunk 1 above)
Comment (MS-TimothyMothra) [reply SDL/BinSkimSetup.ps1]: THAT would make things easier!
But i need to go up three parents. Is there an easier way than this:

```
[string]$buildDirectory = (Join-Path -Path (Split-Path -parent (Split-Path -parent (Split-Path -parent $PSCommandPath))) -ChildPath "bin\Debug\Src") 
```

**Comment by MS-TimothyMothra** [comment]
added script to copy bin DLLs to a new directory for BinSkim scanning tool

**Pull request merged**